TTS/auto_tts/complete_recipes.py

```diff
@@ -0,0 +1,159 @@
+
+        self.data_path = data_path
+        self.batch_size = batch_size
+        self.output_path = output_path
+        self.mixed_precision = mixed_precision
+        self.learning_rate = learning_rate
+        self.epochs = epochs
+        self.model = TtsModels(
+            batch_size=self.batch_size,
+            mixed_precision=self.mixed_precision,
+            learning_rate=self.learning_rate,
+            epochs=self.epochs,
+        )
+
+    def ljspeechAutoTts(
+        self,
+        model_name,
+        tacotron2_model_type=None,
+        glow_tts_encoder=None,
+        forward_attention=False,
+        location_attention=True,
+    ):
+        """This is the auto tts recipe for the ljspeech dataset,
+        current supported models are tacotron2(with ddc and dca),
+        glow tts, vits tts, and speedy speech"""
+        if model_name == "tacotron2":
+            if tacotron2_model_type == "double decoder consistency":
+                dataset, audio = data_loader(
+                    name="ljspeech", path=self.data_path, stats_path="stats_path/scale_stats_ddc.npy"
+                )
+                model_config = self.model.single_speaker_tacotron2_DDC(
+                    audio, dataset, forward_attn=forward_attention, location_attn=location_attention
+                )
+            elif tacotron2_model_type == "dynamic convolution attention":
+                dataset, audio = data_loader(
+                    name="ljspeech", path=self.data_path, stats_path="stats_path/scale_stats_dca.npy"
+                )
+                model_config = self.model.single_speaker_tacotron2_DCA(
+                    audio, dataset, forward_attn=forward_attention, location_attn=location_attention
+                )
+            else:
+                dataset, audio = data_loader(name="ljspeech", path=self.data_path, stats_path=None)
+                model_config = self.model.single_speaker_tacotron2_base(
+                    audio, dataset, forward_attn=forward_attention, location_attn=location_attention
+                )
+        elif model_name == "glow tts":
+            dataset, audio = data_loader(name="ljspeech", path=self.data_path)
+            model_config = self.model.ljspeech_glow_tts(audio, dataset, encoder=glow_tts_encoder)
+        elif model_name == "speedy speech":
+            dataset, audio = data_loader(name="ljspeech", path=self.data_path)
+            model_config = self.model.ljspeech_speedy_speech(audio, dataset)
+        elif model_name == "vits tts":
+            dataset, audio = data_loader(name="ljspeech", path=self.data_path)
+            model_config = self.model.ljspeech_vits_tts(audio, dataset)
+        args, config, output_path, _, c_logger, tb_logger = init_training(TrainingArgs(), model_config)
+        trainer = Trainer(args, config, output_path, c_logger, tb_logger)
+        return trainer
+
+    def vctkAutoTts(self, model_name, speaker_file, glowtts_encoder):
```

Comment:
Maybe rather than defining different functions for each dataset you can make the dataset an argument to the function. AFAIS, the only difference between functions is the choice of datasets.